### ISSUE TITLE
MQAR mamba training

### ISSUE THREAD

**Issue by Moriarty0923** [HazyResearch/zoology#24]
Hello, thank you for your great work on the paper!
We're conducting MQAR experiments on Mamba and we following your dataset construction process.  
We found that the model may overfit on the training set and result in 0 on the test set when we extended the input length and increased the number of kv pairs(4096-256).
We're quite puzzled by this and are wondering if you've ever encountered a similar situation in your experiments.
Could we possibly be missing some important experimental settings that you might have implemented? Any insights or suggestions you could provide would be greatly appreciated.

**Comment by seyuboglu** [comment]
We found that all models struggle to generalize when training only on very long sequences (_e.g._ 4096 tokens). By using a mix of long and short sequences in the training dataset we can generalize to longer sequences. For example: 
https://github.com/HazyResearch/zoology/blob/d90de385ad9febeae8b53d98f105af089ac2a477/zoology/experiments/arxiv24_based_figure2/configs.py#L14
